### ISSUE TITLE
Bloc seems much slower on P12

### ISSUE THREAD

**Issue by plantec** [pharo-graphics/Bloc#557]
no idea where it can come from but the difference is obvious  watching at theses video
(the slower is on p12)

https://github.com/user-attachments/assets/5210eee4-a48b-4cca-a1fe-48477f1040e1

https://github.com/user-attachments/assets/74d17fa8-5237-4c68-8e43-0efe81126243

https://github.com/user-attachments/assets/94fe579e-180c-49cc-8c17-a4179beedd79

https://github.com/user-attachments/assets/ef16b40f-3a8e-424d-bca4-a6fd9dc1f359

**Comment by Rinzwind** [comment]
In the first video, it looks like you have the world renderer canvas scale factor set to 2, while you are using a version of Bloc that does not include the changes of pull request #465. That combination makes the drawing of a BlMorphicSpaceHostMorph slower because the Bloc space form has scale 1 and gets scaled to scale 2 every time it is drawn. You could set the world renderer canvas scale factor to 1 as follows:

```smalltalk
OSWorldRenderer autoSetCanvasScaleFactor: false; canvasScaleFactor: 1
```

But I would suggest to instead update the version of Bloc to one including the changes of pull request #465. Those changes were applied to the ‘Pharo12’ branch as well in pull request #537.

For the following screenshot, I created new Pharo 12 build 1521 images and used the menu item ‘Load Toplo and Bloc’, the image on the left has the world renderer canvas scale factor set to 1 and the image on the right has it set to 2, the text on the right looks better in both the Morphic and Bloc parts (click to zoom in):

<p align="center">
<image width="570" src="https://github.com/user-attachments/assets/795b5f11-b41d-407b-842a-fe762fd5cf07">
</p>

**Comment by plantec** [comment]
Thanks !
Now the BlMorphicSpaceHostMorph works as expected and I can continue to use it :). It is more confortable for running examples.
 Now, the loss of efficiency of Bloc in general still remains.
One can compare with 
ToListElementStresserWithAdditionalSelecters runInSDL

**Comment by labordep** [comment]
Seems similar on my PC, in fact difficult to see the difference (P11 to the right).

https://github.com/user-attachments/assets/3e2dcdae-ce9c-4e51-a548-51e6df933d67

My config:
- Windows 11 Pro
- Intel Core i9-12900K

**Comment by tinchodias** [comment]
FYI, the [BlocBenchs project](https://github.com/pharo-graphics/BlocBenchs/wiki) gathers benchmarking utilities for Bloc, and now there is `BlbSpaceTallyPrinter runToStdout` that filters the results of `SpaceTally` and prints it in a table.

The results show a huge growth of instances during the execution of `ToListElementStresserWithAdditionalSelecters` (a stress demo). When I close the window of this demo and check the space tally again, the table returns to the initial situation. So nothing indicates a memory leak, but we may need to trigger garbage collection passes to wipe some of the tons of instances that are accumulated by this stress demo.

**Comment by tinchodias** [comment]
@plantec do you consider this is valid? I am not sure where we stop debugging.

**Comment by plantec** [comment]
the question is why is it slower in p12 than in p11 ? 
For now we don't have the answer.

**Comment by labordep** [comment]
@tinchodias do you have a difference of performance on your pc between p11 and p12?

**Comment by tinchodias** [comment]
This issue grabbed attention, to me and @tesonep, and we have worked to reproduce it and discover the causes. 

Pablo recently added a stats window to monitor and is working on a refactoring of Bloc benchmarks to help monitoring the performance changes.

The only difference I know was shown by the space tally: for some characteristic of this Toplo stresser demo, the garbage collector in Pharo 12 is not triggered as often as in P11, and some collections with temporal (unreferenced) objects grow too much until they are freed. And those collections become slow when they are so full. So, correct me if I'm wrong but I consider it's not so general as the subject states "bloc much slower in P12".

What should we do to close this issue? what do you think Pablo, Alain, Pierre? or maybe rephrase the subject to be more like "Toplo stresser is slower in P12 vs P11"

**Comment by labordep** [comment]
Thanks @tinchodias for the feedback, I agree.

**Comment by plantec** [comment]
thanks for investigating this. 
I think we can close it.
